File: rust/src/models/one_time_fee.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct OneTimeFee {
    pub quantity: i32,

    pub rate: rust_decimal::Decimal,
}

impl OneTimeFee {
    pub fn new(quantity: i32, rate: rust_decimal::Decimal) -> Self {
        Self { quantity, rate }
    }
}

File: typescript/src/models/oneTimeFee.ts
// this file is @generated

export interface OneTimeFee {
  quantity: number;

  rate: string;
}

export const OneTimeFeeSerializer = {
  _fromJsonObject(object: any): OneTimeFee {
    return {
      quantity: object["quantity"],
      rate: object["rate"],
    };
  },

  _toJsonObject(self: OneTimeFee): any {
    return {
      quantity: self.quantity,
      rate: self.rate,
    };
  },
};
